### ISSUE TITLE
reconnection stopped with channelWrapper event listeners

### ISSUE THREAD

**Issue by ralcini** [benbria/node-amqp-connection-manager#1]
Hi,

we have a strange behaviour when connection-manager retry to reconnect to a RabbitServer and some listener is registered on channelWrapper. In the following test case, channel manager stop trying reconnecting and exit with the following trace 
```
/home/test/lib/node_modules/amqplib/lib/channel.js:148
    throw new IllegalOperationError(msg, stack);
          ^
IllegalOperationError: Channel closed
    at ConfirmChannel.<anonymous> (/home/test/lib/node_modules/amqplib/lib/channel.js:148:11)
    at ConfirmChannel.C.closeBecause (/home/test/lib/node_modules/amqplib/lib/channel.js:200:8)
    at ConfirmChannel.C.close (/home/test/lib/node_modules/amqplib/lib/channel_model.js:77:8)
    at ChannelWrapper._onDisconnect (/home/test/lib/node_modules/amqp-connection-manager/lib/ChannelWrapper.js:81:14)
    at AmqpConnectionManager.<anonymous> (/home/test/lib/node_modules/amqp-connection-manager/lib/ChannelWrapper.js:4:59)
    at AmqpConnectionManager.emit (events.js:117:20)
    at ChannelModel.<anonymous> (/home/test/lib/node_modules/amqp-connection-manager/lib/AmqpConnectionManager.js:88:19)
    at ChannelModel.emit (events.js:95:17)
    at Connection.emit (events.js:95:17)
```

Removing channelWrapper registration, all works as expected.

Test case
```
var amqp = require('amqp-connection-manager');

var connection = amqp.connect(['amqp://test:test@127.0.0.1:5672'], {json: true});
connection.on('connect', function() {
    console.log('Connected!');
});
connection.on('disconnect', function(params) {
    console.log('Disconnected.', params.err.stack);
});

// Set up a channel listening for messages in the queue.
var channelWrapper = connection.createChannel({
    setup: function(channel) {
        return  channel.assertExchange("assertFail");
    }
});

channelWrapper.waitForConnect().then(function() {
    console.log("Listening for messages");
});

channelWrapper.on('connect', function(){console.log("connect")})
channelWrapper.on('error', function(){console.log("error")})
channelWrapper.on('drop', function(){console.log("drop")})
channelWrapper.on('close', function(){console.log("close")})
```

**Comment by jwalton** [comment]
By coincidence, we saw this exact problem on our production server this morning.  Thanks for the test case!

**Comment by jwalton** [comment]
Should be fixed in 1.0.1.

**Comment by ralcini** [comment]
good! =)
